internal/metrics/benchmark_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package benchmark_test
+
+import (
+	"log"
+	"testing"
+
+	prometheus "github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/jaegertracing/jaeger/internal/metrics/otelmetrics"
+	prom "github.com/jaegertracing/jaeger/internal/metrics/prometheus"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	promExporter "go.opentelemetry.io/otel/exporters/prometheus"
+)
+
+func benchmarkCounter(b *testing.B, factory metrics.Factory) {
+	counter := factory.Counter(metrics.Options{
+		Name: "test_counter",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+
+	for i := 0; i < b.N; i++ {
+		counter.Inc(1)
+	}
+}
+
+func BenchmarkPrometheusCounter(b *testing.B) {
+	reg := prometheus.NewRegistry()
+	factory := prom.New(prom.WithRegisterer(reg))
+	benchmarkCounter(b, factory)
+}
+
+func BenchmarkOTELCounter(b *testing.B) {
+	registry := prometheus.NewRegistry()
+	exporter, err := promExporter.New(promExporter.WithRegisterer(registry))
+	if err != nil {
+		log.Fatalf("Failed to create Prometheus exporter: %v", err)
+	}
+	meterProvider := metric.NewMeterProvider(
+		metric.WithReader(exporter),
+	)
+	factory := otelmetrics.NewFactory(meterProvider)
+	benchmarkCounter(b, factory)
+}
```

Comment:
Yes, I think this is correct. The results are awful, I may need to escalate to OTEL Go SIG.
```
$ go test -benchmem -benchtime=5s -bench=Benchmark ./internal/metrics/
goos: darwin
goarch: arm64
pkg: github.com/jaegertracing/jaeger/internal/metrics
BenchmarkPrometheusCounter-10    	816434554	         7.152 ns/op	       0 B/op	       0 allocs/op
BenchmarkOTELCounter-10          	31806798	       186.5 ns/op	      16 B/op	       1 allocs/op
PASS
ok  	github.com/jaegertracing/jaeger/internal/metrics	12.973s
```

Reply:
Yes exactly, I see the results and start questioning my implementation.